components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/workflow/APIStateChangeWorkflow.java

```diff
@@ -63,13 +65,15 @@
     private ApiDAO apiDAO;
     private APISubscriptionDAO apiSubscriptionDAO;
     private APILifecycleManager apiLifecycleManager;
+    private APIGateway apiGateway;
 
     public APIStateChangeWorkflow(ApiDAO apiDAO, APISubscriptionDAO apiSubscriptionDAO, WorkflowDAO workflowDAO,
-                                  APILifecycleManager apiLifecycleManager) {
+                                  APILifecycleManager apiLifecycleManager, APIGateway apiGateway) {
```

Comment:
We are changing the Constructor signature. If this is a minor version release then we need to add a new constructor. For this change, the version release should be a major version change.